src/TorchSharp/NN/Bilinear.cs

```diff
@@ -83,10 +91,22 @@ static public Bilinear Bilinear(long in1Features, long in2Features, long outputS
 
             public static partial class functional
             {
-                static public Tensor bilinear(Tensor x1, Tensor x2, long in1Features, long in2Features, long outputSize, bool hasBias = true)
+                /// <summary>
+                /// Applies a bilinear transformation to the incoming data
+                /// </summary>
+                /// <returns></returns>
+                static public Tensor bilinear(Tensor input1, Tensor input2, Tensor weight, Tensor? bias = null)
                 {
-                    using (var d = nn.Bilinear(in1Features, in2Features, outputSize, hasBias)) {
-                        return d.forward(x1, x2);
+                    var in1Features = input1.shape[input1.shape.Length - 1];
+                    var in2Features = input2.shape[input2.shape.Length - 1];
```

Comment:
NIT: inputX.shape[^1]

Reply:
Good catch. Still getting used to the idea that C# supports this kind of thing...

Reply:
Whenever I use that feature, I have to look up if I need to put `^0` or `^1` to get the last element. I wish it would be clearer.